### DISCUSSION TITLE
OneDrive Client Version v2.5.0-alpha-4 Feedback Tracker

### DISCUSSION THREAD

**Discussion by abraunegg** [abraunegg/onedrive#2570] [General]
Please comment here regarding feedback regarding 'alpha-4'

**Comment by abraunegg** [comment]
1. The initial ETA shows `-596523:-14:-8` when upload progress is at 0%. Expectation: `--:--:--` 
2. Upload percentage is shown with two decimal points `80.00%`. Expectation: `80%`
3. There is a missing whitespace between the dots and "100%". Expectation: `..... 100%`
4. Multi-threading does not seem to work with file uploads. When I drop 3 larger files into a sync_dir while onedrive is running in monitor mode, I can see they are being uploaded one at a time.
5. I'm not sure if resumption of partial downloads works as expected. When I stop monitoring mode during large file download, the partial file is not completed after I start the monitoring mode again.
6. Filename is only being displayed when doing large file download|upload

**Comment by abraunegg** [comment]
@bpozdena
**RE:** 1. The initial ETA shows `-596523:-14:-8` when upload progress is at 0%. Expectation: `--:--:--` 

I have not been able to reproduce this, however this should be fixed now in `v2.5.0-alpha-4 GitHub version: v2.4.25-72-g38155ed`. Please can you help validate this.

**RE:** 2. Upload percentage is shown with two decimal points `80.00%`. Expectation: `80%`

Fixed in `v2.5.0-alpha-4 GitHub version: v2.4.25-80-g8a7a850`

**RE:** 3. There is a missing whitespace between the dots and "100%". Expectation: `..... 100%`

This should be fixed in  `v2.5.0-alpha-4 GitHub version: v2.4.25-72-g38155ed` however the output is going to change again shortly as per above.

**RE:** 5. I'm not sure if resumption of partial downloads works as expected. When I stop monitoring mode during large file download, the partial file is not completed after I start the monitoring mode again.

This was actually never a current v2.5.x *feature* .. resumable downloads was not yet coded or implemented - so this is why this was not working. It is also now tracked as a separate Feature Request: https://github.com/abraunegg/onedrive/issues/2576

**RE:** 6. Filename is only being displayed when doing large file download|upload

This should be fixed in  `v2.5.0-alpha-4 GitHub version: v2.4.25-72-g38155ed` - however I am going to change this again shortly. The full filename path will be used, but it will mean that console logging output will be kind of messy.

**Reply by bpozdena** [reply]
@abraunegg , I've just tested on version `v2.5.0-alpha-4 GitHub version: v2.4.25-74-g6fbf7fa` and all looked great. 

 1 - Fix confirmed
 3 - Fix confirmed
 5 - Thanks for the info
 6 - Fix confirmed. The full file path is also shown correctly even with very long paths. It looks pretty good to me with wide enough terminal window. 

![image](https://github.com/abraunegg/onedrive/assets/24818591/5f43392f-964f-4471-9605-0fbb813f5a41)


BTW: I've started using the alpha build with my real data (150GB, 200k files, 30k directories)  several days ago and my experience has been great so far. No other issues to report.

**Reply by abraunegg** [reply]
Thanks for the feedback

**Comment by abraunegg** [comment]
@bpozdena 
> BTW: I've started using the alpha build with my real data (150GB, 200k files, 30k directories) several days ago and my experience has been great so far. No other issues to report.

Curious on real-world performance, speed, time for a full `--resync` etc ... if you are able to provide any metrics / comparison that would be greatly appreciated.

**Comment by abraunegg** [comment]
@bpozdena 
I have updated the 'download progress tracker output' as indicated above. This now *should* look like this when using `v2.5.0-alpha-4 GitHub version: v2.4.25-80-g8a7a850` or greater.

This version also fixes an upload calculation bug - it *may* have also been contributing to your negative value output - dont know for sure.

### Uploading
```
New items to upload to OneDrive: 5
Total New Data to Upload:        123 MB
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 0%   |  ETA    --:--:--
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 0%   |  ETA    --:--:--
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 0%   |  ETA    --:--:--
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 0%   |  ETA    --:--:--
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 0%   |  ETA    --:--:--
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 33%  |  ETA    00:00:18
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 33%  |  ETA    00:00:22
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 33%  |  ETA    00:00:26
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 66%  |  ETA    00:00:08
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 66%  |  ETA    00:00:08
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 100% | DONE in 00:00:20
Uploading new file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... done.
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 66%  |  ETA    00:00:10
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 100% | DONE in 00:00:20
Uploading new file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... done.
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 100% | DONE in 00:00:22
Uploading new file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... done.
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 33%  |  ETA    00:01:14
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 66%  |  ETA    00:00:20
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 100% | DONE in 00:00:41
Uploading new file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... done.
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 33%  |  ETA    00:01:54
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 66%  |  ETA    00:00:30
Uploading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 100% | DONE in 00:01:02
Uploading new file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... done.
```

### Downloading
```
Creating local directory: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe
Number of items to download from OneDrive: 5
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 0%   |  ETA    --:--:--
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 0%   |  ETA    --:--:--
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 0%   |  ETA    --:--:--
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 0%   |  ETA    --:--:--
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 0%   |  ETA    --:--:--
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 5%   |  ETA    00:00:19
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 10%  |  ETA    00:00:09
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 5%   |  ETA    00:00:19
...
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 95%  |  ETA    00:00:01
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 45%  |  ETA    00:00:10
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 65%  |  ETA    00:00:05
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... 100% | DONE in 00:00:08
Downloading file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file2.data ... done
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... 100% | DONE in 00:00:08
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 40%  |  ETA    00:00:14
Downloading file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file4.data ... done
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 50%  |  ETA    00:00:09
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 70%  |  ETA    00:00:04
...
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 85%  |  ETA    00:00:02
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... 100% | DONE in 00:00:11
Downloading file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file0.data ... done
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 90%  |  ETA    00:00:02
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 75%  |  ETA    00:00:04
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 95%  |  ETA    00:00:01
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 80%  |  ETA    00:00:03
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... 100% | DONE in 00:00:11
Downloading file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file1.data ... done
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 85%  |  ETA    00:00:02
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 90%  |  ETA    00:00:02
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 95%  |  ETA    00:00:01
Downloading: random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... 100% | DONE in 00:00:12
Downloading file random_large_files/J59MWi89p2McuStsvvzXTcH7hckrdTAe/file3.data ... done
```

**Comment by abraunegg** [comment]
@bpozdena 
> 4. Multi-threading does not seem to work with file uploads. When I drop 3 larger files into a sync_dir while onedrive is running in monitor mode, I can see they are being uploaded one at a time.

Regarding this item - how are you dropping the files in?

One at-a-time into a folder? Moving a folder with multiple items into a folder being watched?

In all my test cases - creating the new files individually (using a script) and moving a folder that contains many files - in each instance / case, multiple uploads are being performed at the same time (multi-threading) ..... 

If I can get some clarity here around the issue / claim I can test for this and work out what is potentially going wrong.
